src/ui/transforms.js

```diff
@@ -58,22 +58,22 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
     });
 
     sjrk.storyTelling.transforms.arrayToString = function (input, extraInputs) {
-        var contentString = "";
-        var separator = extraInputs.separator();
-        var stringOnly = extraInputs.stringOnly();
-        var path = extraInputs.path();
+        var combinedString = "";
+        var separator = extraInputs.separator(),
+            stringOnly = extraInputs.stringOnly(),
+            path = extraInputs.path();
 
         fluid.each(input, function (term) {
             if (path) {
                 term = fluid.get(term, path) || term;
             }
 
             if (!stringOnly || (term && typeof term === "string")) {
```

Comment:
Do you need the guard on `term`? I think `typeof(term)` would return `"undefined"` if it didn't exist. Or, are you trying to prevent adding empty strings?

Reply:
Correct, I added that to keep empty strings (well, falsy values, strictly speaking) from being added to the resulting string.

Reply:
This check is a bit vague and given it is prepared for terms to be nonstrings, it should also be prepared to deal with oddities such as finding the number 0. I suggest checking against checking against the empty string explicitly.

Reply:
@amb26 do you mean something like `if (!stringOnly || (typeof term === "string" && term !== ""))`?

By the current logic, if it's the number 0 not wrapped in a string, then it will be excluded if `stringsOnly` is true, and if that option is false then `0` will be allowed. I suppose I could add some checks for falsy or for truthy but meaningless values like:
- the empty array
- the empty object
- `"0"` (the string)
- `"false"` (the string)

I feel that at this point, my goal is to put only the things that convey information into a human-readable string, i.e. something that looks a bit like natural language without all the accoutrement of curly braces and double-quotes. That said, if I intend to merge this into Infusion some day, these restrictions may be too much and we might want to keep it broad and allow falsy or "meaningless" values to be stringified as-is.

I'm also unsure about how to handle complex objects, given `JSON.stringify` will include all of the object structure and layout. This would make full inversion more easily possible, but readability worse.